### ISSUE TITLE
[trezor] User can't spend out from approved tspend

### ISSUE THREAD

**Issue by alexlyp** [decred/decrediton#3678]
User reported seeing this error when attempting to spend an outpoint from an approved tspend while in Trezor mode.

https://github.com/decred/decrediton/blob/caca418f8a76ddcda9d3612166010af52069d662/app/helpers/trezor.js#L43

They mentioned Exodus having a similar issue so probably related to not handling the tspends properly.

**Comment by chappjc** [comment]
What happened in the Exodus case?  Asking in case there's a dcrdata change to be made

**Comment by matheusd** [comment]
I guess that the underlying trezor firmware impl won't be able to recognize tspends due to their outputs being special and therefore it won't be able to spend them.

In particular, tspend outputs are not _exactly_ standard outputs (they have an OP_TGEN tag at the start) and so it's likely the trezor firmware won't recreate the tspend to be able to spend from it.

cc @JoeGruffins

**Comment by JoeGruffins** [comment]
Yes sorry for the delay. Current firmware cannot spend  treasury type outputs.

**Comment by JoeGruffins** [comment]
I will get this working with custom firmware and create an issue then pr with trezor to make this possible with Trezor T. It should be a simple change*.

**Comment by JoeGruffins** [comment]
Have started the process to get this in the firmware: https://github.com/trezor/trezor-firmware/issues/2096
